### PR TITLE
[ADVAPP-706]:  Create report for the report library for Prospects

### PR DESCRIPTION
**Pull request by jayushi-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-706

### Technical Description

> Create report in report library for Prospects module

### Any deployment steps required?

>No.

### Are any Feature Flags Added?

> No.

_______________________________________________

### CODE REVIEW

**Comment by jayushi-canyon** [comment]
> Failing test as we discussed. Please re-request the review when it is ready.
@Orrison All test are successful, you can review the PR.

File: app-modules/report/src/Filament/Widgets/ProspectReportStats.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                maxPrecision: 2,
+            )),
+            Stat::make('Total Segments', Number::abbreviate(
+                Cache::tags([$this->cacheTag])->remember('alert-count', now()->addHours(24), function (): int {
+                    return Segment::query()->whereModel(SegmentModel::Prospect)->count();
+                }),
+                maxPrecision: 2,
+            )),
+            Stat::make('Total Tasks', Number::abbreviate(
+                Cache::tags([$this->cacheTag])->remember('alert-count', now()->addHours(24), function (): int {
```
Comment (Orrison) [inline app-modules/report/src/Filament/Widgets/ProspectReportStats.php:78]:
```suggestion
                Cache::tags([$this->cacheTag])->remember('prospect-tasks-count', now()->addHours(24), function (): int {
```

File: app-modules/report/src/Filament/Widgets/ProspectReportLineChart.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                'y' => [
+                    'min' => 0,
+                ],
+            ],
+        ];
+    }
+
+    protected function getData(): array
+    {
+        $runningTotalPerMonth = Cache::tags([$this->cacheTag])->remember('saved_conversations_line_chart', now()->addHours(24), function (): array {
```
Comment (Orrison) [inline app-modules/report/src/Filament/Widgets/ProspectReportLineChart.php:71]:
```suggestion
        $runningTotalPerMonth = Cache::tags([$this->cacheTag])->remember('total-prospects-line-chart', now()->addHours(24), function (): array {
```

File: app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Report\Filament\Widgets;
+
+use Filament\Tables\Table;
+use Livewire\Attributes\On;
+use Filament\Tables\Columns\TextColumn;
+use AdvisingApp\Prospect\Models\Prospect;
+use Filament\Widgets\TableWidget as BaseWidget;
```
Comment (Orrison) [inline app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php:43]: Is this rename needed? Nothing else is called `TableWidget` in this class or referenced here?

If not, we should just use it normally.

File: app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                    return Prospect::whereIn($key, function ($query) use ($key) {
+                        $query->select($key)
+                            ->from((new Prospect())->getTable())
+                            ->orderBy('created_at', 'desc')
+                            ->take(100);
+                    })->orderBy('created_at', 'desc');
+                }
+            )
+            ->defaultSort('created_at', 'desc')
+            ->columns([
```
Comment (Orrison) [inline app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php:97]: Let's remove `searchable` from each of these columns. We don't need this page to be searchable

File: app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                TextColumn::make('full_name')
+                    ->label('Name')
+                    ->searchable(),
+                TextColumn::make('email')
+                    ->searchable(),
+                TextColumn::make('status.name')
+                    ->badge()
+                    ->translateLabel()
```
Comment (Orrison) [inline app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php:105]: This `translateLabel` is not needed here.

File: app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+          //   //   ->orderBy('created_at', 'desc')
+          //   //   ->take(100)
+
+          //   // Prospect::whereIn('id', function ($query) {
+          //   //   $query->select('id')
+          //   //     ->from(with(new Prospect())->getTable())
+          //   //     ->orderBy('created_at', 'desc')
+          //   //     ->take(100);
+          //   // })
+          // )
```
Comment (Orrison) [inline app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php:83]: Is this commented code needed anymore?

File: resources/views/advising-prospect/filament/pages/prospect-report.blade.php
Comment (Orrison) [inline resources/views/advising-prospect/filament/pages/prospect-report.blade.php:1]: I don't think this file is needed.

Comment (jayushi-canyon) [reply resources/views/advising-prospect/filament/pages/prospect-report.blade.php:1]: Removed

File: app-modules/report/src/Filament/Widgets/ProspectReportStats.php [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            )),
+            Stat::make('Total Alerts', Number::abbreviate(
+                Cache::tags([$this->cacheTag])->remember('alert-count', now()->addHours(24), function (): int {
+                    return Alert::whereHasMorph('concern', Prospect::class)->count();
+                }),
+                maxPrecision: 2,
+            )),
+            Stat::make('Total Segments', Number::abbreviate(
+                Cache::tags([$this->cacheTag])->remember('alert-count', now()->addHours(24), function (): int {
+                    return Segment::query()->whereModel(SegmentModel::Prospect)->count();
```
[the remark this answers is not in the document]
Comment (jayushi-canyon) [reply app-modules/report/src/Filament/Widgets/ProspectReportStats.php]: Done

File: app-modules/report/src/Filament/Widgets/ProspectReportStats.php [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+                maxPrecision: 2,
+            )),
+            Stat::make('Total Alerts', Number::abbreviate(
+                Cache::tags([$this->cacheTag])->remember('alert-count', now()->addHours(24), function (): int {
+                    return Alert::whereHasMorph('concern', Prospect::class)->count();
+                }),
+                maxPrecision: 2,
+            )),
+            Stat::make('Total Segments', Number::abbreviate(
+                Cache::tags([$this->cacheTag])->remember('alert-count', now()->addHours(24), function (): int {
```
[the remark this answers is not in the document]
Comment (jayushi-canyon) [reply app-modules/report/src/Filament/Widgets/ProspectReportStats.php]: Done

File: app-modules/report/src/Filament/Widgets/ProspectReportStats.php [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        return [
+            Stat::make('Total Prospects', Number::abbreviate(
+                Cache::tags([$this->cacheTag])->remember('prospects-count', now()->addHours(24), function (): int {
+                    return Prospect::count();
+                }),
+                maxPrecision: 2,
+            )),
+            Stat::make('Total Alerts', Number::abbreviate(
+                Cache::tags([$this->cacheTag])->remember('alert-count', now()->addHours(24), function (): int {
```
[the remark this answers is not in the document]
Comment (jayushi-canyon) [reply app-modules/report/src/Filament/Widgets/ProspectReportStats.php]: Done

File: app-modules/report/src/Filament/Widgets/ProspectReportLineChart.php (hunk 2 above)
Comment (jayushi-canyon) [reply app-modules/report/src/Filament/Widgets/ProspectReportLineChart.php]: Done

File: app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php (hunk 3 above)
Comment (jayushi-canyon) [reply app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php]: Done

File: app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php (hunk 4 above)
Comment (jayushi-canyon) [reply app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php:84]: Done

File: app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php (hunk 5 above)
Comment (jayushi-canyon) [reply app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php]: Done

File: app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php (hunk 6 above)
Comment (jayushi-canyon) [reply app-modules/report/src/Filament/Widgets/ProspectReportTableChart.php]: Removed

**Review by Orrison — CHANGES_REQUESTED** [review]
@jayushi-canyon I think you need to rebase this branch onto `main` to get the most recent changes. As it seems that there are 300+ files changes in this PR now related to files underneath `.composer/cache` which is a file that is now in the `.gitignore` in `main`
